submissions/andrewklmn/friends-app/app.js

```diff
@@ -0,0 +1,358 @@
+  
+  if (state.scrollDisabled) return true;
+
+  if (container.scrollTop + container.clientHeight >= --container.scrollHeight) {
+    if(numberOfShowedFriends < friends.length) {
+      state.scrollDisabled = true;
+      container.scrollTop--;
+      preloader.classList.remove('hidden');
+
+      const additionalFriendList = sortList(filterList(state))
+        .slice( numberOfShowedFriends, numberOfShowedFriends + initialListLength);
+
+      requestAnimationFrame(() => drawMoreFriends(additionalFriendList) );
+
+      setTimeout(()=>{    
+        state.scrollDisabled = false;
+        preloader.classList.add('hidden');
+      }, nextMoreFriendAutoloadDelay);
+    };
+  }
+};
+
+const sorterKeyChangeHandler = (target, state) => {
+  const {sorter} = state;
+  if (target.value === '') {    
+    sortOrder.innerHTML = '';
+    sorter.keyName = '';
+    sorter.order = '';
+  } else {
+    if(sortOrder.innerHTML === ''){
+      sortOrder.innerHTML = sorter.orderSymbols.ASC;
+      sorter.order = 'ASC';
+    };    
+    sorter.keyName = target.value; 
+  };
+  redrawFriends(state);
+}
+
+const sorterOrderClickHandler = (target, state)=>{
+  const {sorter} = state;
+
+  if(sortField.value === ''){
+    target.innerHTML = '';
+    sorter.order = '';
+    return;
+  };
+  if (sorter.order === 'ASC') {
+    target.innerHTML = sorter.orderSymbols.DESC;
+    sorter.order = 'DESC';
+  } else {
+    target.innerHTML = sorter.orderSymbols.ASC;
+    sorter.order = 'ASC';
+  };
+
+  redrawFriends(state);
+}
+
+const searchFieldChangeHandler = (target, state)=>{
+  const {filters} = state;
+  if (filters.partOfName === target.value) {
+    return true;
+  }
+  filters.partOfName = target.value;
+  redrawFriends(state);
+}
+
+const filterGenderChangeHandler = (target, state) => {
+  const {filters} = state;
+  if (target.value === filters.gender) {
+    return true;
+  }
+  filters.gender = target.value;
+  redrawFriends(state);
+}
+
+const MIN_AGE_RANGE_INDEX = 0; 
+const MAX_AGE_RANGE_INDEX = 1;
+
+const filterAgeChangeHandler = (target, state, ageRangeIndex)=>{
+  state.filters.ageRange[ageRangeIndex] = target.value;
+  redrawFriends(state);
+}
+
+const filterCountryChangeHandler = (target, state) => {
+  const {filters} = state;
+  if (target.value === filters.country) {
+    return true;
+  }
+  filters.country = target.value;
+  redrawFriends(state);
+}
+
+document.addEventListener('DOMContentLoaded',() => {
+  initApp(state);
+  
+  sortField.addEventListener('change', ({target}) => sorterKeyChangeHandler(target, state));
+  sortOrder.addEventListener('click', ({target}) => sorterOrderClickHandler(target, state));
+  
+  resetButton.addEventListener('click', () => resetButtonClickHandler(state));
+  
+  searchField.addEventListener('keyup',({target}) => searchFieldChangeHandler(target, state));
+  
+  filterGender.addEventListener('change', ({target}) => filterGenderChangeHandler(target, state));
```

Comment:
I'd recommend you combining these `change` event handlers here into one handler. It can help you to get rid of the functions with the similar logic and increase the performance of your app.